src/vec.cpp

```diff
@@ -1644,13 +1644,14 @@ grid_volume grid_volume::subvolume(ivec is, ivec ie) {
 
 void grid_volume::init_subvolume(ivec is, ivec ie) {
   ivec origin(dim, 0);
+  for (int i=0;i<3;i++) num[i] = 0;
   LOOP_OVER_DIRECTIONS(dim, d) {
-    num[(int)d] = (ie - is).in_direction(d) / 2;
+    num[(int)d] = (ie - is).in_direction(d)/2;
     origin.set_direction(d, is.in_direction(d));
   }
   num_changed();
-  center_origin();
-  shift_origin(origin);
+  //center_origin();
+  set_origin(origin);
```

Comment:
This is the last bit of code that needs to work in order for this PR to pass (I hope).

I am using `subvolumes` to pull the proper `dft` array index given a specified `ivec` (using the `grid_volume::index()` function). Unfortunately, the `subvolume` code was never tested or debugged.

The current issue is that the `index()` routine is not supplying the proper index...

The `strides`, and `num` arrays are set properly, which means that the issue has to be with the `io` field (the origin of the `subvolume()`.

Any thoughts?

Reply:
From the `grid_volume::index` method, it seems like the origin should be set to `is - iyee_shift(c)` for your use case.